demo/json_demo.py

```diff
@@ -0,0 +1,52 @@
+"""This file provides preliminary methods of accessing and storing JSON data."""
+import json
+import os
+
+
+def main():
+    """Pull data from a provided json, and write it back to the json."""
+    user_data_dict = get_dict_from_json_file("demofile")
+    print("Dictionary:\n")
+    for username in user_data_dict:
+        print(username + ":", user_data_dict[username], "\n")
+    json_format_string = write_dict_to_json_file(user_data_dict, "demofile")
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+    new_user = {"testuser": {"commits": [], "issues": [], "pull_requests": []}}
+    json_format_string = add_user_to_json(
+        user_data_dict, new_user, "testuser", "demofile"
+    )
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+
+
+def get_dict_from_json_file(json_file_name):
+    """Populate and return a dictionary of all the data in a specified json file."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "r") as json_file:
+        # In the open() function, "r" specifies read-only access
+        json_string = json_file.read()
+        user_data_dict = json.loads(json_string)
+        # json.loads() converts a json-formatted string into a python dictionary
+    return user_data_dict
+
+
+def write_dict_to_json_file(user_data_dict, json_file_name):
+    """Overwrite specified json file with data from a given dictionary."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "w") as json_file:
+        # In the open() function, "w" specifies write access
+        json_string = json.dumps(user_data_dict, indent=4)
+        # json.dumps() converts a dictionary into a json-formatted string
+
+        # Specifying an indent does not alter the data itself, it only increases
+        # readability in the json file.
+
+        json_file.write(json_string)
+    return json_string
```

Comment:
The main reason I return the string is so that a test case can check the string to see if it matches the text that was actually put into the file. The only thing that serves to test is essentially that the json package works as intended. Should I still make this change and remove the seemingly redundant code? I am under the impression dump() does not return a string that I then could return for the test case I mentioned.

Reply:
Yes it is more testable if containing a return statement to some extent. But even if with the return string, the only testable thing is that whether json package correctly convert a `dict` to a `JSONstring`, but not checking if it is correctly written to the file, which is the main purpose of this method. Using either function, the way to test if it is correctly written to file is probably to read from the file and see if the expected data is being written. I'm still more leaning to using `dump`. This is a pretty trivial issue, though, so I guess either way is fine at this point.